### PR TITLE
Update test image for e2e-framework

### PR DESCRIPTION
**Pull request by ShwethaKumbla**
#### What this PR does 
Update test image to kubekins which has docker and kind tool.

/cc @vladimirvivien @cpanato

### CODE REVIEW

File: config/jobs/kubernetes-sigs/e2e-framework/e2e-framework-presubmits.yaml [hunk 1]
```diff
@@ -6,7 +6,7 @@ presubmits:
     path_alias: sigs.k8s.io/e2e-framework
     spec:
       containers:
-      - image: gcr.io/k8s-staging-releng/releng-ci:latest-go1.16
+      - image: gcr.io/k8s-testimages/kubekins-e2e:v20210615-c973edd-master
```
Comment (cpanato) [inline config/jobs/kubernetes-sigs/e2e-framework/e2e-framework-presubmits.yaml:9]: did you test if that will work?

Comment (vladimirvivien) [reply config/jobs/kubernetes-sigs/e2e-framework/e2e-framework-presubmits.yaml:9]: @cpanato is there an easy way to test the image outside of prawl ?
cc @ShwethaKumbla

Comment (cpanato) [reply config/jobs/kubernetes-sigs/e2e-framework/e2e-framework-presubmits.yaml:9]: here: https://github.com/kubernetes/test-infra/blob/master/prow/build_test_update.md#How-to-test-a-ProwJob

Comment (cpanato) [reply config/jobs/kubernetes-sigs/e2e-framework/e2e-framework-presubmits.yaml:9]: we can merge this and try out as well :)

Comment (ShwethaKumbla) [reply config/jobs/kubernetes-sigs/e2e-framework/e2e-framework-presubmits.yaml:9]: No. I am not sure how to test this changes in local. If you can suggest me how can I verify this change that would be great.

We need a kind cluster to test our code (Unit tests). I tried installing kind but it has dependency with docker and in the current images docker is not configured.  I observed most of the other projects use `kubekins` image so I updated the image with kubekins.

Comment (cpanato) [reply config/jobs/kubernetes-sigs/e2e-framework/e2e-framework-presubmits.yaml:9]: hum, then maybe we need to setup the privileged to true for this job, can you please check other jobs that uses this image to see how they are setup?

Comment (ShwethaKumbla) [reply config/jobs/kubernetes-sigs/e2e-framework/e2e-framework-presubmits.yaml:9]: Yes, I see other jobs which use kubekins configured with privileged to true and command as runner.sh

          securityContext:
            privileged: true

          command:
              # generic runner script, handles DIND, bazelrc for caching, etc.
              - runner.sh

File: config/jobs/kubernetes-sigs/e2e-framework/e2e-framework-presubmits.yaml [hunk 2]
```diff
@@ -22,11 +25,14 @@ presubmits:
...
         imagePullPolicy: Always
         command:
+        - runner.sh
```
Comment (cpanato) [inline config/jobs/kubernetes-sigs/e2e-framework/e2e-framework-presubmits.yaml:31]: I think should be something like
```
- command:
   - runner.sh
    args:
    - make
    - test
```

File: config/jobs/kubernetes-sigs/e2e-framework/e2e-framework-presubmits.yaml [hunk 3]
```diff
@@ -6,11 +6,15 @@ presubmits:
...
```
Comment (cpanato) [inline config/jobs/kubernetes-sigs/e2e-framework/e2e-framework-presubmits.yaml:12]: for this job we can keep the previous image, this is running lint and other checks, we don't need to have special things for this

**Review by cpanato — APPROVED** [review]
/lgtm

**Review by vladimirvivien — APPROVED** [review]
/approve

**Pull request merged**